src/types/external/duration.rs

```diff
@@ -0,0 +1,21 @@
+use chrono::Duration;
+use iso8601_duration as iso8601;
+
+use crate::{InputValueError, InputValueResult, Scalar, ScalarType, Value};
+
+/// Implement the Duration scalar
+///
+/// The input/output is a string in ISO8601 format.
+#[Scalar(internal, name = "Duration")]
```

Comment:
There is a new `specified_by_url` field specified by the Graphql 2021 spec to describe the current implementation:

```suggestion
#[Scalar(internal, name = "Duration", specified_by_url = "https://en.wikipedia.org/wiki/ISO_8601#Durations")]
```